src/ray/rpc/gcs_server/gcs_rpc_client.h

```diff
@@ -46,10 +46,14 @@ class Executor {
 };
 
 // Define a void GCS RPC client method.
+// Timeout can be configured in 3 levels; whole service, handler, and each call.
+// The priority of timeout is each call > handler > whole service
+// (the lower priority timeout is overwritten by the higher priority timeout).
 #define VOID_GCS_RPC_CLIENT_METHOD(SERVICE, METHOD, grpc_client, method_timeout_ms,    \
                                    SPECS)                                              \
   void METHOD(const METHOD##Request &request,                                          \
```

Comment:
do you think it's a good idea to add something like with promise future set?
```
void SYNC_METHOD(...)
```

Reply:
Code-wise, it will be probably cleaner in the long term, and I think the performance of additional copy is probably not that big deal for synchronous calls (in the first place, if the RPC is synchronous, I believe the performance shouldn't be that important). But we can think about ways to optimizing it later as well.